### DISCUSSION TITLE
Dry run option in terratest

### DISCUSSION THREAD

**Discussion by Syed-Shahidh-Ilhan** [gruntwork-io/knowledge-base#811] [Help] [answered]
Hey guys, I'm just exploring how terratest can help me test individual modules.
so in terratest we're actually spinning up the infrastructure running some checks and then destroying it.....There must be some costs due to this right? (like spinning up an eks cluster and then destroying it)
So is there a dry-run option in terratest?

---

<ins datetime="2024-03-28T19:04:32Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110849">Tracked in ticket #110849</a></p>
</ins>

**Comment by denis256** [comment, marked as answer]
Hello,
examples of how to use terratest can be found at https://github.com/gruntwork-io/terratest/tree/master/examples

Some of examples can run locally and will not require access to cloud resources:
 * docker tests
 * Helm
 * Terraform tests can be configured to create local files 

https://github.com/gruntwork-io/terratest/tree/master/examples/docker-hello-world-example

https://github.com/gruntwork-io/terratest/tree/master/examples/helm-basic-example

https://github.com/gruntwork-io/terratest/tree/master/examples/terraform-hello-world-example
